src/SAML2/XML/ExtensionPointInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\SAML2\XML;
+
+/**
+ * Interface for several extension points objects.
+ *
+ * @package simplesamlphp/saml2
+ */
+interface ExtensionPointInterface
+{
+    /**
+     * Get the local name for the element's xsi:type.
+     *
+     * @return string
+     */
+    public static function getXsiTypeName(): string;
+
+
+    /**
+     * Get the namespace for the element's xsi:type.
+     *
+     * @return string
+     */
+    public static function getXsiTypeNamespaceURI(): string;
+
+
+    /**
+     * Get the namespace-prefix for the element's xsi:type.
+     *
+     * @return string
+     */
+    public static function getNamespacePrefix(): string;
+
+
+    /**
+     * Return the xsi:type value corresponding this element.
+     *
+     * @return string
+     */
+    public static function getXsiType(): string;
```

Comment:
Shouldn't we get rid of this one? It's kind of redundant now, since we have methods to get both the prefix and the CName separately, and then a bad implementation could lead to inconsistencies.

Reply:
Fine with me.. I'll drop it